File: learnAPI/4.r
library(jsonlite)
library(dplyr)
library(httr)
library(tibble)
library(tidyr)

url <- "http://universities.hipolabs.com/search?name=binus&country=Indonesia"

response <- GET(url)

if (http_status(response)$category == "Success") {
    json_content <- content(response, "text")
    data <- fromJSON(json_content)

    df <- as.data.frame(data)
    print(head(df))
}

File: learnAPI/4.py
import requests
import pandas as pd

url = "http://universities.hipolabs.com/search?country=United+States"

response = requests.get(url)

if response.status_code == 200:
    data = response.json()
    
    df = pd.DataFrame(data)
    df['domains'] = df['domains'].apply(lambda x: ''.join(x))
    df['web_pages'] = df['web_pages'].apply(lambda x: ''.join(x))
    print(df)
else:
    print("Failed to retrieve data:", response.status_code)